ftplugin/purescript_pscide.vim

```diff
@@ -346,28 +377,49 @@ function! s:goToDefinition(definedAt)
   endif
 endfunction
 
-function! PSCIDErebuild(stuff)
+function! PSCIDErebuild(async, ...)
   let g:psc_ide_suggestions = {}
   let filename = expand("%:p")
   let input = {'command': 'rebuild', 'params': {'file': filename}}
 
-  let resp = s:callPscIde(input, 0, 0)
+  if a:0 > 0 && type(a:1) == v:t_func
```

Comment:
Yes but that's useful if you want to define your own callback.  The syntastic plugin does not support async build processes, hence it blocks on writing the file.  I supply my custom function here.  I will make separate pull request to include it, after this patch gets in.

Expert from my `.vim/after/ftplugin/purescript.vim` file:
```viml
function! s:PSCIDEerrors(llist)
  let qflist = []
  for e in a:llist
    let eparts = split(e, ":")
    call add(
	  \   qflist
	  \ , { "bufnr": bufnr(eparts[1])
	  \   , "filename": eparts[1]
	  \   , "lnum": eparts[2]
	  \   , "col": eparts[3]
	  \   , "text": eparts[4]
	  \   , "type": eparts[0]
	  \   }
	  \ )
  endfor
  call setqflist(qflist)
endfunction
if g:psc_ide_syntastic_mode == 0
  com! PSCIDErebuild call PSCIDErebuild(1)
  augroup purescript
    au! BufWritePost *.purs call PSCIDErebuild(1, function("s:PSCIDEerrors"))
  augroup END
endif
```